src/Aquifer.Public.API/Endpoints/Bibles/Texts/Get/Endpoint.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Aquifer.Common.Utilities;
+using Aquifer.Data;
+using Aquifer.Data.Entities;
+using FastEndpoints;
+using Microsoft.EntityFrameworkCore;
+
+namespace Aquifer.Public.API.Endpoints.Bibles.Texts.Get;
+
+public class Endpoint(AquiferDbContext dbContext) : Endpoint<Request, Response>
+{
+    private readonly AquiferDbContext _dbContext = dbContext;
+    public override void Configure()
+    {
+        Get("/bibles/{BibleId}/texts");
+        Description(d => d.ProducesProblemFE());
+        Summary(s =>
+        {
+            s.Summary = "Gets the Bible text contained within a book of the Bible.";
+            s.Description =
+                "For a given Bible and book of the Bible, returns the Bible text (and optional audio information if available) for all verses within the chapter and verse parameters.";
+        });
+    }
+
+    public override async Task HandleAsync(Request request, CancellationToken ct)
+    {
+        var startChapter = request.StartChapter ?? 1;
```

Comment:
I disagree with this approach. Just set the default value on the prop. Who cares if it was set in the request or not? This just clutters up the code with extra variables and extra lines.

Reply:
Some thoughts in reply:
* Yes, it takes an extra line in the endpoint here but I removed a line from the Request (the attribute specifying the default added to each Request property) so it's breaking even on line/code count.
* Putting the default value here helps to put the business logic in one place here in the endpoint instead of having the request contain knowledge about default values.
* The `Validator` is where having the nullability was most useful because I could tell if the user passed a value.
* Nullability exists for cases like this where data may or may not be provided.  This accurately represents the data given to us over the wire.

Reply:
Anther thought: In this case there is a difference in meaning between `null` and `0` or `999` for these start/end parameters.  We only filter text if the user asks for filtering by supplying these parameters and have validation in place to ensure that they are specified together or not at all.  Using non-nullable defaults make it harder to check if the user supplied a value or not.  For example, if the user supplies a start chapter and end chapter of `1` and also `5` and `999` for start verse and end verse, respectively, which is a legitimate request, then we will say the request is invalid because the validation will have to assume that because `999` is the default value that the user didn't supply it in the request (even though they did).

It looks like other validation in [Resource search](https://github.com/BiblioNexusStudio/aquifer-server/blob/4918bd0293f47df24b56b0ed120067af89ab4d0a/src/Aquifer.Public.API/Endpoints/Resources/Search/GetResources/Validator.cs#L35-L38) gets around this problem by assuming that `0` is essentially equivalent to `null` but it still fails to validate consumers passing `0`.  I'd expect https://qa.api.aquifer.bible/resources/search?languageCode=eng&limit=100&bookCode=gen&startVerse=0&endVerse=0&api-key=741ad467d648470487e0a853e2eb7f64 to return an error but it ignores the consumer input because it has to assume that the consumer didn't pass those `startVerse` and `endVerse` values when doing validation.

Reply:
I don't know what you mean about the validation. It checks that it's greater than or equal to 1 in all cases, whether they pass it or not that validation wouldn't change. If the user passes 0 then it fails and tells them they need to pass 1 or greater. Also don't understand your reasoning around the situation where the user passes 999 for end verse that that would fail. The validator works off the request object and has the default values. If they pass it then the values are what they pass, if they don't, it's what the default is. Whether they pass 999 or not the validation succeeds.

I wouldn't even put default values in the docs on this because it's optional and doesn't matter. The explanation for the fields should say anyway that if they're not provided it will grab the whole book or whole chapter whatever.

By doing this we're allocating four new variables that don't do anything so it's inefficient, not to mention the extra overhead of wrapping int in Nullable (we're gettin' pedantic now!).

It's confusing as a reader, because now I have to try and understand why the request values aren't being used and have a default value being set to a different variable instead of just having a default themselves.

Again I come back to the question of who cares if the user supplied it or not as long as they get the right data? Why does it matter if they sent 1 for the start verse or didn't send anything for the start verse? The result is the same. No one is looking at these request payloads. And saying nullability exists for reasons like this is also why default values exist. We generally don't want things to be null, and if there's a sensible default value, we should use it to avoid having to deal with null. Null is generally a bad thing and creates bugs. There's a reason you have to explicitly call out your strings as nullable now. Nullability introduces the possibility for bugs, and should be avoided as much as possible. There are people who even go to such extremes as wrapping all their objects in a NotNull object. My experience has always been that null is a bad thing and you should avoid it as much as reasonable.

The thing with the business logic doesn't really resonate with me, since it having a null value default is just as much business logic as it having a default of 1. Should every bool be nullable so that it doesn't have a default value of false? Should every int be nullable so it doesn't have a default value of 0? If a model's default value is business logic, then is the model itself business logic that should be deleted? The model dictates what the default values are for itself, whether that's null or something else. Same with collections, I would always make a collection default empty, rather than null, because it makes the code safer. That's not business logic, that's just defensive coding.

Reply:
> Should every bool be nullable so that it doesn't have a default value of false?

Yes, that's the logical conclusion of my argument.  I've done just that in the past for query string parameters and request body parameters.  `null` means "not passed" and non-null means "provided by the consumer."  URL path variables, however, are not optional because they are required in order to invoke the correct route handler and thus they can't be null.

> Null is generally a bad thing and creates bugs.

I agree with avoiding `null` in the cases where `null` doesn't have meaning.  I'm a big fan of the new(er) non-nullable types for that reason (e.g. `string` vs `string?`) because the compiler helps you to avoid worrying about `null` states.

> Again I come back to the question of who cares if the user supplied it or not as long as they get the right data?

I don't think they will get the correct response.  I'll go back to the pre-existing resources search route on the Public API (https://qa.api.aquifer.bible/resources/search?languageCode=eng&limit=100&bookCode=gen&startVerse=0&endVerse=0&api-key={key}) for my example.  I'd argue that this should return an error response to the consumer instead of returning the entire chapter.  But with current defaults of `0` it's not possible to do that.  You could, say, change the defaults to `-1` instead of `0` but then if a consumer passes `-1` for start verse and end verse then they'll get the entire chapter instead of an error message. There's no way to solve this bug without differentiating between "the user passed data" and "the user didn't pass data" which I don't see being possible with default values on non-nullable `int`s for those parameters.

There's an argument that we're fine with this bug and it's worth that weird edge case in order to have simpler code.  But I don't think I'm making the code that more complex or incurring significant performance hits and the compiler kindly reminds me with warnings if I forget to check for `null` which prevents bugs.

Reply:
I think it's fair that the search example is a little weird because it treats 0 as a starting point. But it's also a bit of a different situation because start and end verse being null is actually a valid value. So I could see the argument for those actually being nullable, because verse references aren't required for finding a resource. However, in the case of the Bible text endpoint, they are required to pull the correct text.

There's no bug though. The validator specifically allows 0 as an input in the search. In the case of this request, we wouldn't be allowing 0 as a valid value, so someone passing startVerse=0 would return an error that they need to pass a valid value between 1 and 999. So I'm unclear still on how you say that we must know if the user passed a value or not, because whether they pass 1 or they pass nothing and the default is 1, they get the same result.

Overall though, I'm just not hearing a reason for why we would sacrifice simplicity and cleaner code for something that has no impact on the correctness of the system. I think it's still coming down to the question for me of why we care if the user passed the default or didn't, when the result is the same.

Reply:
> So I'm unclear still on how you say that we must know if the user passed a value or not, because whether they pass 1 or they pass nothing and the default is 1, they get the same result.

Ok.  I looked at the public API for validation examples for the start/end chapter and verse parameters and found [the public API's Resource search validation](https://github.com/BiblioNexusStudio/aquifer-server/blob/4918bd0293f47df24b56b0ed120067af89ab4d0a/src/Aquifer.Public.API/Endpoints/Resources/Search/GetResources/Validator.cs#L30-L41).  This validation is more complicated and does things like requiring start verse to be present if end verse is also present and requiring start chapter to be present if start verse is present.

It sounds like I should not copy that example and should instead do something more simple like the internal API's [bible text validation](https://github.com/BiblioNexusStudio/aquifer-server/blob/4918bd0293f47df24b56b0ed120067af89ab4d0a/src/Aquifer.API/Endpoints/Bibles/Texts/Get/Validator.cs#L12-L13) which only confirms that `start <= end` (and probably add in the `InclusiveBetween(1, 999)`?

Reply:
I would, yeah.

Reply:
I switched to using defaults instead of nullable here: [commit](https://github.com/BiblioNexusStudio/aquifer-server/pull/443/commits/915feb0d16be4d3215c483cc56b3e3975e5ae498).